l10n_nl_xaf_auditfile_export/models/xaf_auditfile_export.py

```diff
@@ -203,6 +203,13 @@ def _get_auditfile_template(self):
 
     def button_generate(self):
         """Generate, store and validate auditfile."""
+        # First check wether file is already there. Should not be possible, because of
+        # the locking, but there is a corner case for single threaded servers, where
+        # this button can be pressed after another sessions has already started the
+        # generation. Then this method will only run after the first job released the
+        # lock and finished processing.
+        if self.auditfile:
+            raise exceptions.UserError(_("Auditfile has already been generated."))
```

Comment:
Right, but in that single-threaded case, just let it be regenerated a second time.

This case is IMO worse because if you want to regenerate the auditfile you can't?

Reply:
Already you can't. The generate button is disabled when auditfile is filled. So it is not taking away functionality.